채널 페이지 렌더링 최적화 시도

aereeeee opened the pull request:
# 채널 페이지 렌더링 최적화 시도

## 해당 이슈 📎

- 채널페이지의 슬라이드컴포넌트 렌더링 최적화 (#132)

## 변경 사항 🛠

- 채팅에서 불필요하게 컨텍스트 데이터를 사용하고 있어 제거함
- 슬라이드 리사이즈 이벤트에 불필요한 렌더링을 줄이기 위해 throttle 적용

happydhKim commented on web/src/components/channel/Chat/Chat.jsx:
```diff
@@ -7,7 +7,7 @@ import ChatSort from './ChatSort';
 
 const ChatToggle = (props) => {
-  const { channelId, userId } = props;
+  const { channelId, userId, slideLength } = props;
```
3개니까 내려주세욥!😕

happydhKim commented on web/src/components/channel/Chat/Chat.jsx:
```diff
@@ -28,33 +28,45 @@ const ChatToggle = (props) => {
 
 const Chat = (props) => {
-  const { channelId, userId } = props;
+  const { channelId, userId, slideLength } = props;
```
3개니까 내려주세욥!😕

happydhKim commented on web/src/components/channel/Chat/ChatInput/ChatInput.jsx:
```diff
@@ -26,13 +25,12 @@ const ChatInput = (props) => {
   const { isAnonymous } = useGetUserStatus();
-  const { channelId, setQuestionToggle } = props;
-  const limit = useChannelSelector((state) => state.slideUrls.length);
+  const { channelId, setQuestionToggle, slideLength } = props;
```
여기두용!!☺️

happydhKim commented on web/src/components/channel/Chat/ChatLogs/ChatCards/ChatCards.jsx:
```diff
@@ -5,7 +5,7 @@ import ChatCard from './ChatCard';
 
 const ChatCards = (props) => {
-  const { userId, chats } = props;
+  const { userId, chats, slideLength } = props;
```
여기두용!!☺️

happydhKim commented on web/src/components/channel/Chat/ChatLogs/ChatLogs.jsx:
```diff
@@ -32,7 +32,7 @@ const scrollCommander = {
 const ChatLogs = (props) => {
   const scrollWrapRef = useRef(null);
-  const { userId, questionToggle } = props;
+  const { userId, questionToggle, slideLength } = props;
```
여기두용!!☺️

happydhKim reviewed (approved):
마지막 날까지 고생하셨어요!!!!!!!🐤

aereeeee committed:
[ REFACTOR ] 채널 페이지 렌더링 개선 (#132)

Why
-채팅 컴포넌트에서 불필요하게  컨텍스트 상태를 사용하고 있어 렌더링이 자주 일어난다.

What
- 컨텍스트 데이터를 제거하고 프롭으로 데이터를 전달하여 처리함
